### PR TITLE
Docs/designing-section

### PR DESCRIPTION
**Pull request by mighdoll**
doc section on principles for designing extensions

### CODE REVIEW

**Review by k2d222 — APPROVED** [review]
Good overview of our design choices for new contributors 👍

File: Designing.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+As experiments move toward stability
+we'll move experimental enhancements into more tools behind a feature flag
+so more projects can try them.
+
+## Words and Reserved Words in WGSL
+
+We'll typically want to avoid using keywords used in current WGSL, unless
+the meaning is consistent and additive to the keyword's use in WGSL.
+
+We'll want to avoid conflicting with the meaning of keywords in future
```
Comment (stefnotch) [inline Designing.md:86]: We could get away without this requirement.

1. Whenever WGSL itself gets a new syntactical construct, all the WESL tools need to be updated anyways
2. With the editions mechanism, it is possible to completely change WESL without breaking any of the consumers

We probably still want at least a relaxed version of this.

Comment (mighdoll) [reply Designing.md:86]: we'll have the technical means to translate anything w/o conflict, I agree!

The remaining question is what to do if there's a likely future confusion.. On the one hand, if we're prototyping a wgsl feature than might just get adopted as is, it'd be better for everyone if we use a reserved word. But if we're expecting the meaning of the keyword to be different in a future rev, then better to use a different word.
  
I'll try an edit..

Comment (stefnotch) [reply Designing.md:86]: That sounds like a very reasonable option, I approve of this.

**Pull request merged**